rholangweb/rholang/main/templates/index.html

```diff
@@ -80,13 +85,56 @@ <h3 class="panel-title">Output</h3>
             <div class="tab-content">
               <p role="tabpanel" class="tab-pane" id="rbl"
                    ><code>{{ rbl_code }}</code></p>
+              {% if run_error %}
+              <pre role="tabpanel" class="tab-pane active alert alert-warning" id="repl"
+               >{{ run_error }}</pre>
+              {% else %}
               <pre role="tabpanel" class="tab-pane active" id="repl"
                    >{{ repl_session }}</pre>
+              {% endif %}
             </div>
           </div>
           {% endif %}
         </div>
     </form>
+
+    <!-- Save Dialog -->
+    <div class="modal fade" id="saveDialog" tabindex="-1"
+         role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
+      <div class="modal-dialog" role="document">
+        <div class="modal-content">
+          <div class="modal-header">
+            <h5 class="modal-title" id="saveAs">Save Rholang to File</h5>
+            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+              <span aria-hidden="true">&times;</span>
+            </button>
+          </div>
+          <form action="save" method="POST">
+            {% csrf_token %}
+            <div class="modal-body">
+              <label>Filename: <input name="basename" />.rho</label>
+              <input type="hidden" name="rhoSave" id="rhoSave" />
+            </div>
+            <div class="modal-footer">
+              <button type="button" class="btn btn-secondary" data-dismiss="modal"
+                      >Cancel</button>
+              <input type="submit" class="btn btn-primary" id="save" data-dismiss="modal"
+                     value="Save" />
+            </div>
+          </form>
+        </div>
+      </div>
+    </div>
+    <hr />
+    <address class="footer">
+      Copyright (c) 2017-2018 <a href="https://rchain.coop">RChain Cooperative</a>
+      <br />
+      <small>
+        built with <a href="https://www.djangoproject.com/">Django</a>
+        <br />
+        See <a href="django-license">Django license</a>.
```

Comment:
Related to my above comment on the license, I think we should just link directly to https://github.com/django/django/blob/master/LICENSE .

Reply:
I suppose RHOL-103 doesn't say exactly how the link should work, but since we're including the license in our code, I thought we might as well link to our copy.